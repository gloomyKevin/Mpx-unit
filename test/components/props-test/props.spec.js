const { loadComponent } = require('../../util')
const simulate = require('miniprogram-simulate')

describe('props component test', () => {
  it('render with props test', () => {
    const id = loadComponent('src/components/props-test.mpx')
    const comp = simulate.render(id, { propsContent: 'Hello' })
    comp.attach(document.createElement('parent-wrapper'))
    expect(comp.toJSON()).toMatchSnapshot()
  })

  // it('render without props test', () => {
  //   const id = loadComponent('src/components/props-test.mpx')
  //   const comp = simulate.render(id, {})
  //   comp.attach(document.createElement('parent-wrapper'))
  //   expect(comp.toJSON()).toMatchSnapshot()
  // })
})
