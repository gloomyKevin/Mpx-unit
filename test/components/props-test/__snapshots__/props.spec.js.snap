// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<main>
  <wx-view
    style=""
  >
    Hello
  </wx-view>
</main>
`;

exports[`props component test render props test 1`] = `
<main>
  <wx-view
    style=""
  >
    Hello
  </wx-view>
</main>
`;

exports[`props component test render with props test 1`] = `
<main>
  <wx-view
    style=""
  >
    Hello
  </wx-view>
</main>
`;
